db/migrate/20201112214758_create_contact_attempts.rb

```diff
@@ -0,0 +1,43 @@
+class CreateContactAttempts < ActiveRecord::Migration[6.0]
+  def up
+    create_table :contact_attempts do |t|
+      t.references :patient, index: true
+      t.references :user, index: true
+
+      t.boolean :successful, index: true
+      t.string :note
+
+      t.timestamps
+    end
+
+    # mysql does not support full joins
+    execute <<-SQL.squish
+      INSERT INTO contact_attempts (patient_id, user_id, successful, note, created_at, updated_at)
+      SELECT histories.patient_id as patient_id,
+            users.id as user_id,
+            histories.comment NOT LIKE '%unsuccessful%' as successful,
+            SUBSTRING_INDEX(histories.comment, 'uccessful contact attempt. Note: ', -1) as comment,
+            histories.created_at as created_at,
+            histories.updated_at as updated_at
+      FROM histories
+      LEFT JOIN users
+      ON histories.created_by = users.email
+      WHERE history_type = 'Contact Attempt'
+      UNION
```

Comment:
I'm confused why everything starting at this UNION and below is even necessary given what is above which looks almost the same except for the UNION and RIGHT JOIN? I am not super well versed in SQL yet so I may just be missing something.

Reply:
so the whole point of this UNION thing is to make this a FULL JOIN rather than an INNER JOIN, and this is to ensure that this query includes histories where the 'created_by' field doesn't match a user's email (if the user's email was changed or some other case)

Reply:
Ah - well in that case wouldn't we not have a `user_id` to set here even with the FULL JOIN which is already problematic for creating the `ContactAttempt`?

I know I pointed out above just for awareness, but I don't actually think we need to account for that case here given that at this point, ALL contact attempt history items will have to have been triggered by a user.

Reply:
ah I guess that's true, I just wanted to play it safe in case someone had changed their email or something and the lookup no longer matches, in that case, I think the history item wouldn't show up

Reply:
That's fair. It's probably okay -  I was just thinking about how long it might take to run this on prod.

Reply:
yeah that's a totally fair concern, and something that I potentially see as something that could take a while to execute as well. I think the `LIKE` keyword on the unindexed `comment` column is what's going to make this take a long time, but I can't think of any way to get around it

Reply:
Yeah understood. I'll resolve this for now and we can see how fast it is on demo with about 5k monitorees.